Detectors/TPC/workflow/src/CATrackerSpec.cxx

```diff
@@ -371,12 +380,65 @@ DataProcessorSpec getCATrackerSpec(ca::Config const& specconfig, std::vector<int
         activeSectors |= sectorHeader->activeSectors;
         validInputs.set(sector);
         datarefs[sector] = ref;
-        if (caClusterer) {
-          inputDigits[sector] = pc.inputs().get<gsl::span<o2::tpc::Digit>>(ref);
+        if (caClusterer && !zs) {
+            inputDigits[sector] = pc.inputs().get<gsl::span<o2::tpc::Digit>>(ref);
           LOG(INFO) << "GOT SPAN FOR SECTOR " << sector << " -> " << inputDigits[sector].size();
         }
       }
+      if (zs){
+    	  sizes = {0};
+    	  // sizes: #zs pages per endpoint:
+          std::vector<InputSpec> filter = {{"check", ConcreteDataTypeMatcher{gDataOriginTPC, "ZSSIZES"}, Lifetime::Timeframe}};
+          for (auto const& ref : InputRecordWalker(pc.inputs(), filter)) {
+      	    std::cout << "get zs sizes from processing context" << std::endl;
+            sizes = pc.inputs().get<std::array<unsigned int, NEndpoints*NSectors>>(ref);
+          }
+          // zs pages
+          std::vector<InputSpec> filter2 = {{"check", ConcreteDataTypeMatcher{gDataOriginTPC, "TPCZS"}, Lifetime::Timeframe}};
+          for (auto const& ref : InputRecordWalker(pc.inputs(), filter2)) {
+    	    std::cout << "get zs input from processing context" << std::endl;
+    	    //std::string inputLabel = "zsinput";
+            inputZS = pc.inputs().get<gsl::span<ZeroSuppressedContainer8kb>>(ref);
+        }
+        //set all sectors as active and as valid inputs
+        for (int s = 0; s < NSectors; s++){
+		  activeSectors |= 1 << s;
+		  validInputs.set(s);
+        }
+        /*
+		if (verbosity > 1) {
+		    std::vector<const ZeroSuppressedContainer*> pagePtr;
+		    pagePtr.resize(inputZS.size());
+			std::cout << "Span size: " << inputZS.size() << std::endl;
+			for (unsigned int i = 0; i < inputZS.size(); i++ ){
+			    pagePtr[i] =  reinterpret_cast<const ZeroSuppressedContainer*>(&inputZS[i]);
+			    std::cout << "Page: " << i << " Version: " << (int)pagePtr[i]->hdr.version << " timebins: " << (int)pagePtr[i]->hdr.nTimeBins << " CRUID: "<< (int)pagePtr[i]->hdr.mCRUID << " time offset: " << (int)pagePtr[i]->hdr.mtimeOffset << " n ADC samples: " << (int)pagePtr[i]->hdr.nADCsamples << std::endl;
+			  }
+		}*/
```

Comment:
There seems to be a lot of outcommented debug code, which I think should be removed from the PR. The CATrackerSpec.cxx file is already quite long.

Reply:
I agree to remove the debug stuff before merging. Currently I still marked this as draft, since the output to digits is not yet final and there is the need to debug further.